src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -343,7 +347,77 @@ public void Stop()
             }
         }
 
+
+        /* W3C support functionality (see https://w3c.github.io/trace-context) */
+
+        /// <summary>
+        /// Holds the W3C 'tracestate' header.   This is typically used by logging systems
+        /// to store vendor-specific information that must flow.
+        /// Logically it is just a kind of baggage (if flows just like baggage), but because
+        /// it is expected to be special cased (it has its own HTTP header), it is more 
+        /// convenient/efficient if it is not lumped in with other baggage.   
+        /// </summary>
+        public string TraceState
+        {
+            get
+            {
+                for (var activity = this; activity != null; activity = activity.Parent)
+                {
+                    var val = activity._traceState;
+                    if (val != null)
+                        return val;
+                }
+                return null;
+            }
+            set
+            {
+                _traceState = value;
+            }
+        }
+  
+        /// <summary>
+        /// Activity tries to use the same format for IDs as its parent it has a parent.  
+        /// However if the activity has no parent, it has to do something.   
+        /// This sets it so that WC3 format is used in this case.  
+        /// </summary>
+        static public bool UseW3CFormat { get; set; }
```

Comment:
it should be set to true by default.
Assuming it's false but some incoming requests have W3C headers, we'll create mixed w3c and 'legacy' activities - this would create an inconsistent experience and not going to help much.

I realize that anyone can change it to make it right, but by default let's be consistent.

Reply:
The default is definitely an interesting question.    My concern is compatibility.  Maybe we just break things, but normally we go through a procedure where we support both for a while first with the default the old way, then the default the new way (with a app compat config that will flip it back), and then get rid of the old way.   

I note that even if the default was off, if things come in from the outside (the most common case), with W3C headers, the code WILL do the right thing.   Thus flipping the default is that THAT important.   The question is how important is current compat.    Generally we find out that it is more important than we first think (people did build on what we shipped).  

I would like to defer this decision, until we get to a later phase of the checkin.